static/app/utils/profiling/colors/utils.tsx

```diff
@@ -0,0 +1,208 @@
+import {Frame} from '../frame';
+
+import {Color, FlamegraphTheme, LCH} from './../flamegraph/FlamegraphTheme';
+
+const uniqueBy = <T,>(arr: ReadonlyArray<T>, predicate: (t: T) => unknown): Array<T> => {
+  const cb = typeof predicate === 'function' ? predicate : (o: T) => o[predicate];
+
+  return [
+    ...arr
+      .reduce((map, item) => {
+        const key = item === null || item === undefined ? item : cb(item);
+
+        if (key === undefined || key === null) return map;
+        map.has(key) || map.set(key, item);
+
+        return map;
+      }, new Map())
+      .values(),
+  ];
+};
+
+// https://en.wikipedia.org/wiki/HSL_and_HSV#From_luma/chroma/hue
+export const fract = (x: number): number => x - Math.floor(x);
+export const triangle = (x: number): number => 2.0 * Math.abs(fract(x) - 0.5) - 1.0;
+export function fromLumaChromaHue(L: number, C: number, H: number): Color {
+  const hPrime = H / 60;
+  const X = C * (1 - Math.abs((hPrime % 2) - 1));
+  const [R1, G1, B1] =
+    hPrime < 1
+      ? [C, X, 0]
+      : hPrime < 2
+      ? [X, C, 0]
+      : hPrime < 3
+      ? [0, C, X]
+      : hPrime < 4
+      ? [0, X, C]
+      : hPrime < 5
+      ? [X, 0, C]
+      : [C, 0, X];
+
+  const m = L - (0.3 * R1 + 0.59 * G1 + 0.11 * B1);
+
+  return [clamp(R1 + m, 0, 1), clamp(G1 + m, 0, 1), clamp(B1 + m, 0, 1.0)];
+}
+
+export const makeStackToColor = (
+  fallback: [number, number, number, number]
+): FlamegraphTheme['COLORS']['STACK_TO_COLOR'] => {
+  return (
+    frames: ReadonlyArray<Frame>,
+    colorMap: FlamegraphTheme['COLORS']['COLOR_MAP'],
+    colorBucket: FlamegraphTheme['COLORS']['COLOR_BUCKET']
+  ) => {
+    const colors = colorMap(frames, colorBucket);
+
+    const length = frames.length;
+
+    // Length * number of frames * color components
+    const colorBuffer: number[] = new Array(length * 4 * 6);
+
+    for (let index = 0; index < length; index++) {
+      const c = colors.get(
+        frames[index].name + (frames[index].file ? frames[index].file : '')
+      );
+      const colorWithAlpha = c ? [...c, 1] : fallback;
```

Comment:
Yes correct. It was a mistake in the design to allow for 3 channel colors, I probably should have just went with 4 channel and leave it as is. This will be something to simplify/optimize later on, the obv win here is that we could be allocating 4bytes less memory (25% less in total) for that alpha channel when it is not required.